py3status/modules/arch_updates.py

```diff
@@ -97,31 +101,36 @@ def _check_pacman_updates(self):
         This method will use the 'checkupdates' command line utility
         to determine how many updates are waiting to be installed via
         'pacman -Syu'.
+        Returns: None if unable to determine number of pending updates
         """
-        if not self.include_pacman:
-            return 0
-        pending_updates = str(subprocess.check_output(["checkupdates"]))
-        return pending_updates.count(LINE_SEPARATOR)
+        try:
+            pending_updates = str(subprocess.check_output(["checkupdates"]))
+            return pending_updates.count(LINE_SEPARATOR)
+        except subprocess.CalledProcessError:
+            return None
 
     def _check_aur_updates(self):
         """
         This method will use the 'cower' command line utility
         to determine how many updates are waiting to be installed
         from the AUR.
+        Returns: None if unable to determine number of pending updates
         """
         # For reasons best known to its author, 'cower' returns a non-zero
         # status code upon successful execution, if there is any output.
         # See https://github.com/falconindy/cower/blob/master/cower.c#L2596
-        if not self.include_aur:
-            return '?'
-
         pending_updates = b""
         try:
-            pending_updates = str(subprocess.check_output(["cower", "-u"]))
+            str(subprocess.check_output(["cower", "--update"]))
+            # Unsuccessful
+            return None
```

Comment:
Can you `confirm` we will `return None` here? It seems strange to have two `return None`.

Reply:
So cower is weird -- it returns non-zero if the command is successful. (If there are 0 updates available, then cower returns 1.) Since `subprocess` raises an exception if the return value is non-zero, if we hit this code, it means the return value was 0 and that there was an error running cower. It's very possible that this code will never be hit.

Reply:
On my side, this one gets hit when we're offline. Maybe it's the other `return None` below that never get hit so I think it would be okay to remove it. I don't think we need `str()` here too.

Reply:
Refactored it a bit to simplify it.